public/app/features/scenes/core/SceneDataProviderNode.tsx

```diff
@@ -0,0 +1,155 @@
+import { cloneDeep } from 'lodash';
+import React from 'react';
+// import { PanelData } from '@grafana/data';
+import { Unsubscribable } from 'rxjs/internal/types';
+
+import { DataQueryExtended, SceneQueryRunner } from '../querying/SceneQueryRunner';
+
+import { SceneDataObject, SceneObjectBase } from './SceneObjectBase';
+import { SceneTimeRange } from './SceneTimeRange';
+import { SceneDataState, SceneLayoutState, SceneObject, SceneObjectStatePlain } from './types';
+
+type SceneDataProviderInputParams = {
+  timeRange: SceneTimeRange;
+} & {
+  [key: string]: SceneObject;
+};
+
+export interface SceneDataProviderNodeState extends SceneObjectStatePlain, SceneLayoutState, SceneDataState {
+  queries: DataQueryExtended[];
+  inputParams: SceneDataProviderInputParams;
+}
+
+export class SceneDataProviderNode extends SceneDataObject<SceneDataProviderNodeState> {
+  static Component = SceneDataProviderRenderer;
+  private runner: SceneQueryRunner = new SceneQueryRunner({ queries: [] });
+  private querySub?: Unsubscribable;
+  private staticQueries: DataQueryExtended[] = [];
+
+  private buildStaticQueries() {
+    const staticQueries: DataQueryExtended[] = [];
+
+    for (let i = 0; i < this.state.queries.length; i++) {
+      const query = this.state.queries[i];
+      const staticQuery = cloneDeep(query);
+
+      for (let prop in query) {
+        if (Object.prototype.hasOwnProperty.call(query, prop)) {
+          console.log(prop, query[prop]);
+          if (query[prop] instanceof SceneObjectBase) {
+            const node = query[prop];
+            console.log('updating', prop, node.state.value);
+            staticQuery[prop] = node.state.value;
+          }
+        }
+      }
+
+      staticQueries.push(staticQuery);
+    }
+
+    return staticQueries;
+  }
```

Comment:
In this experiment queries can contain `SceneObjects`. This function makes sure that the queries passed to the runner have all values resolved according to the current state.

Reply:
what's the use case for queries that can contain SceneObjects?